### PR TITLE
ZEN-28116: filter components by their production state for config

### PR DESCRIPTION
**Pull request by rderkach**
MW wasn't working for Component Group.
Some of the objects n the component group may not
have production state attribute, skip them.
When creating device config for cllector daemon
filter components by their production state in the
same way as devices are.

### CODE REVIEW

**Comment by webhookbotuser** [comment]
Build finished.

File: Products/ZenModel/MaintenanceWindow.py [hunk 1]
```diff
@@ -555,6 +555,11 @@ def setProdState(self, state, ending=False, batchSize=None,
 
         def _setProdState(devices_batch):
             for device in devices_batch:
+                # In case we try to move Component Group into MW
+                # some of objects in CG may not have production state
+                # we skip them.
+                if not hasattr(device, "privateattr_productionState"):
```
Comment (johnlhamilton) [inline Products/ZenModel/MaintenanceWindow.py]: We should not reference privateattr_productionState directly.  Also, what are the cases where this is true?  Most components will NOT have a production state, but will acquire their device's production state.

**Comment by webhookbotuser** [comment]
Build finished.

**Comment by webhookbotuser** [comment]
Build finished.

**Comment by webhookbotuser** [comment]
Build finished.

**Comment by webhookbotuser** [comment]
Build finished.

**Pull request merged**